### PR TITLE
Split services into packages and start them separately

### PR DESCRIPTION
**Pull request by wilzbach**
- starts to move services into their own packages
- starts to separate the core app runtime and services
- makes `http` a global service (Option 1 of https://github.com/storyscript/runtime/issues/27) that can subscribe to route requests from multiple apps
- starts the `http` service still on the main server (easier for development + setup), but already communicates over the wire

With this PR the communication between the `http` service and the runtime executor is via REST requests, though this is just to explore and PoC.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,6 +4,8 @@ FROM ubuntu:18.04
 ENV PORT         3000
 ENV NVM_VERSION  0.35.2
 ENV NODE_VERSION 11.15.0
+ENV RUNTIME_URL   http://localhost:3000
+ENV RUNTIME_HTTP_URL http://localhost:3000/services/http
```
Comment (wilzbach) [inline Dockerfile]: Should be changed to auto-discovery soon, but for two services this is the easiest.

File: package.json [hunk 2]
```diff
@@ -5,32 +5,20 @@
...
     "python-bridge": "^1.1.0",
-    "routing-controllers": "^0.8.0",
-    "simple-oauth2": "^3.3.0",
     "slack": "^11.0.2",
-    "tmp": "^0.1.0",
-    "tmp-promise": "^2.0.2",
-    "twitter": "^1.7.1",
-    "uuid": "^3.4.0",
-    "winston": "^3.2.1"
+    "twitter": "^1.7.1"
```
Comment (wilzbach) [inline package.json]: TODO: these dependencies should only be installed if required by a listed dependency.

Comment (JeanBarriere) [reply package.json]: What about a monorepo for these services @wilzbach, I think we'd need to externalize packages for types but that should be about it ?

File: src/app/events.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+const uuidv1 = require("uuid/v1");
+
+// TODO: replace by DB storage with cache (simple K,V mapping)
+class EventRegistry {
+  // events indexed by id
+  events: { [P: string]: string };
```
Comment (colelawrence) [inline src/app/events.ts]: You could just name `P: string` as `eventID: string`

**Comment by williammartin** [comment]
I think the tests might be failing right now because of changes to the compiler/hub.

File: package.json (hunk 2 above)
Comment (wilzbach) [reply package.json]: Yes, it's just that at the moment when I made this PR this didn't work, hence my comment.
The dependencies are removed now: https://github.com/storyscript/runtime/pull/29/commits/6297add8ff0ed92645d112ec8f2cae7d870c895f

**Comment by wilzbach** [comment]
Superseded by https://github.com/storyscript/runtime/pull/29

**Pull request closed without merging**